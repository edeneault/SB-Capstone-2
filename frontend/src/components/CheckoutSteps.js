import React from "react";
import { Nav, Col, Row } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";

const CheckoutSteps = ({ step1, step2, step3, step4 }) => {
  return (
    <Row>
      <Col xs={12}>
        <Nav className='justify-content-md-center mb-4'>
          <Nav.Item className='mx-1 px-1'>
            {step1 ? (
              <LinkContainer to='/login'>
                <Nav.Link>Sign In</Nav.Link>
              </LinkContainer>
            ) : (
              <Nav.Link disabled>Sign In</Nav.Link>
            )}
          </Nav.Item>

          <Nav.Item className='mx-1 px-1'>
            {step2 ? (
              <LinkContainer to='/shipping'>
                <Nav.Link>Shipping</Nav.Link>
              </LinkContainer>
            ) : (
              <Nav.Link disabled>Shipping</Nav.Link>
            )}
          </Nav.Item>

          <Nav.Item className='mx-1 px-1'>
            {step3 ? (
              <LinkContainer to='/payment'>
                <Nav.Link>Payment</Nav.Link>
              </LinkContainer>
            ) : (
              <Nav.Link disabled>Payment</Nav.Link>
            )}
          </Nav.Item>

          <Nav.Item className='mx-1 px-1'>
            {step4 ? (
              <LinkContainer to='/placeorder'>
                <Nav.Link>Place Order</Nav.Link>
              </LinkContainer>
            ) : (
              <Nav.Link disabled>Place Order</Nav.Link>
            )}
          </Nav.Item>
        </Nav>
      </Col>
    </Row>
  );
};

export default CheckoutSteps;
