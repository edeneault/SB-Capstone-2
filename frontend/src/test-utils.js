import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
