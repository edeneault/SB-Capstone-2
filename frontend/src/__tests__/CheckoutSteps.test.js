import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState, render } from "../test-utils";

import CheckoutSteps from "../components/CheckoutSteps";

const mockStore = configureMockStore([thunk]);

test("renders CheckoutSteps component without crashing", () => {
  const store = mockStore(initialState);

  const { getByText } = render(
    <Provider store={store}>
      <CheckoutSteps />
    </Provider>,
  );

  expect(getByText("Shipping")).toBeInTheDocument();
});

test("matches snapshot", function () {
  const store = mockStore(initialState);
  const { asFragment } = render(
    <Provider store={store}>
      <CheckoutSteps />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
