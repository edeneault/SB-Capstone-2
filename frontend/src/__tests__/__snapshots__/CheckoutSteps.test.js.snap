// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-12"
    >
      <div
        class="justify-content-md-center mb-4 nav"
      >
        <div
          class="mx-1 px-1 nav-item"
        >
          <a
            aria-disabled="true"
            class="nav-link disabled"
            href="#"
            role="button"
            tabindex="-1"
          >
            Sign In
          </a>
        </div>
        <div
          class="mx-1 px-1 nav-item"
        >
          <a
            aria-disabled="true"
            class="nav-link disabled"
            href="#"
            role="button"
            tabindex="-1"
          >
            Shipping
          </a>
        </div>
        <div
          class="mx-1 px-1 nav-item"
        >
          <a
            aria-disabled="true"
            class="nav-link disabled"
            href="#"
            role="button"
            tabindex="-1"
          >
            Payment
          </a>
        </div>
        <div
          class="mx-1 px-1 nav-item"
        >
          <a
            aria-disabled="true"
            class="nav-link disabled"
            href="#"
            role="button"
            tabindex="-1"
          >
            Place Order
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
